docs/filemanagement.rst

```diff
@@ -208,6 +229,54 @@ like so:
 
     content_string = file4.GetContentString(encoding='ISO-8859-1')
 
+Upload data as bytes in memory buffer
+--------------------------------------
+
+Data can be kept as bytes in an in-memory buffer when we use the io module’s
+Byte IO operations, we can upload files that reside in memory, for
+example we have a base64 image, we can decode the string and upload it to drive
+without the need to save as a file and use `SetContentFile(filename)`_
+
+.. code-block:: python
+
+    import io
+    from pydrive2.drive import GoogleDrive
+
+    # create GoogleDrive instance with authenticated GoogleAuth instance.
+    drive = GoogleDrive(gauth)
+    # define file name and type
+    metadata = {
+        'title': 'image_test',
+        'mimeType': 'image/jpeg'
+    }
+    # create file
+    file = drive.CreateFile(metadata=metadata)
+    # Buffered I/O implementation using an in-memory bytes buffer.
+    image_file = io.BytesIO(image_bytes)
+    # set the content of the file
+    file.content = image_file
+    # upload the file to google drive
+    file.Upload()
+
+Upload file to specific folder
+--------------------------------------
+
+In order to upload a file into a specific drive folder we need to pass the
+id of the folder in the metadata param from `CreateFile()`_.
+Save the image from the previous example into a specific folder
+
+.. code-block:: python
+
+    metadata = {
+        'parents': [
+            {"kind": "drive#fileLink", "id": id_drive_folder}
```

Comment:
Q: why do we need `kind` here?

Reply:
i think it not explicitly neccesary but i added to reduce error prone.
[docs](https://developers.google.com/drive/api/v3/reference/files)
`kind` : Identifies what kind of resource this is. Value: the fixed string "drive#file".

Reply:
> Value: the fixed string "drive#file".

hmm, in this case it's `fileLink`

> i think it not explicitly neccesary but i added to reduce error prone.

from the docs it should be level above? not inside `parents`? if it's not needed, let's remove it.

Reply:
`parents` is doing the work here, what we could do is change to:
```python
    metadata = {
        'parents': [id_drive_folder],
        'title': 'image_test',
        'mimeType': 'image/jpeg'
    }
````

as the docs says:
`parents: The IDs of the parent folders which contain the file.
If not specified as part of a create request, the file will be placed directly in the user's My Drive folder. If not specified as part of a copy request, the file will inherit any discoverable parents of the source file. Update requests must use the addParents and removeParents parameters to modify the parents list.`

i will have to test it tho, because im using the `fileLink` version, what do you think?

Reply:
I think it's better to have a simple version of it. Right now it's not clear if it's correct or not (specifying the link this way) and it's needed or not.

Reply:
Hi Ivan, i made some tests
this way is not working for me 
```python
    metadata = {
        'parents': [id_drive_folder],
        'title': 'image_test',
        'mimeType': 'image/jpeg'
    }
```
this way is working
```python
    metadata = {
        'parents': [{'id': id_drive_folder}],
        'title': 'image_test',
        'mimeType': 'image/jpeg'
    }
````